tranx/dataset2.py

```diff
@@ -116,10 +116,13 @@ def preprocess_conala_dataset(train_file, test_file, grammar_file, src_freq=3, c
     print('Avg action len: %d' % np.average(action_lens), file=sys.stderr)
     print('Actions larger than 100: %d' % len(list(filter(lambda x: x > 100, action_lens))),
           file=sys.stderr)
-
-    pickle.dump(train_examples,
-                open(os.path.join(out_dir, 'train.all_{}.bin'.format(num_mined)), 'wb'))
-    pickle.dump(full_train_examples, open(os.path.join(out_dir, 'train.gold.full.bin'), 'wb'))
+    dump_size = int(len(train_examples)/n_files)
+    for i in range(n_files-1):
+        pickle.dump(train_examples[i * dump_size: (i+1) * dump_size],
+                open(os.path.join(out_dir, 'train.all_{}.bin'.format(i)), 'wb'))
```

Comment:
In Python 3.6+ you can write `f'train.all_{i}.bin'` instead of calling format explicitly.
Read more here: https://realpython.com/python-f-strings/